src/ResourceManager/RecoveryServices/Commands.RecoveryServices.Backup.Providers/Providers/AzureFilesPsBackupBackupProvider.cs

```diff
@@ -57,7 +61,328 @@ public class AzureFilesPsBackupProvider : IPsBackupProvider
+                ValidateAzureFilesModifyProtectionRequest(itemBase, policy);
+
+                Dictionary<UriEnums, string> keyValueDict = HelperUtils.ParseUri(item.Id);
+                containerUri = HelperUtils.GetContainerUri(keyValueDict, item.Id);
+                protectedItemUri = HelperUtils.GetProtectedItemUri(keyValueDict, item.Id);
+                sourceResourceId = item.SourceResourceId;
+            }
+
+            // construct Service Client protectedItem request
+
+            AzureFileshareProtectedItem properties = new AzureFileshareProtectedItem();
+
+            properties.PolicyId = policy.Id;
+            properties.SourceResourceId = sourceResourceId;
+
+            ProtectedItemResource serviceClientRequest = new ProtectedItemResource()
+            {
+                Properties = properties
+            };
+
+            return ServiceClientAdapter.CreateOrUpdateProtectedItem(
+                containerUri,
+                protectedItemUri,
+                serviceClientRequest,
+                vaultName: vaultName,
+                resourceGroupName: resourceGroupName);
+        }
+
+        private void ValidateAzureFilesWorkloadType(CmdletModel.WorkloadType type)
+        {
+            if (type != CmdletModel.WorkloadType.AzureFiles)
+            {
+                throw new ArgumentException(string.Format(Resources.UnExpectedWorkLoadTypeException,
+                                            CmdletModel.WorkloadType.AzureFiles.ToString(),
+                                            type.ToString()));
+            }
+        }
+
+        private void ValidateAzureFilesWorkloadType(CmdletModel.WorkloadType itemWorkloadType,
+            CmdletModel.WorkloadType policyWorkloadType)
+        {
+            ValidateAzureFilesWorkloadType(itemWorkloadType);
+            ValidateAzureFilesWorkloadType(policyWorkloadType);
+            if (itemWorkloadType != policyWorkloadType)
+            {
+                throw new ArgumentException(string.Format(Resources.UnExpectedWorkLoadTypeException,
+                                            CmdletModel.WorkloadType.AzureFiles.ToString(),
+                                            itemWorkloadType.ToString()));
+            }
+        }
+
+        private void ValidateFileShareEnableProtectionRequest(string fileShareName, string storageAccountName,
+            string storageAccountResourceGroupName, PolicyBase policy)
+        {
+            if (string.IsNullOrEmpty(fileShareName))
+            {
+                throw new ArgumentException(string.Format(Resources.InvalidFileShareName));
+            }
+            if (string.IsNullOrEmpty(storageAccountName) && string.IsNullOrEmpty(storageAccountResourceGroupName))
+            {
+                throw new ArgumentException(
+                    string.Format(Resources.BothStorageAccountNameAndStorageAccountResourceGroupNameShouldNotEmpty)
```

Comment:
nit: this should be BothStorageAccountNameAndStorageAccountResourceGroupNameShouldNot**Be**Empty